File: selenium/src/main/java/pages/DashboardPage.java
package pages;

import base.BasePage;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class DashboardPage extends BasePage {

    // Locators
    private final By menuButton = By.cssSelector("button[aria-label='Open menu']");
    private final By signOutButton = By.xpath("//button[contains(., 'Sign Out')]");
    private final By financialHealthTitle = By.xpath("//*[contains(text(), 'Financial Health')]");

    public DashboardPage(WebDriver driver) {
        super(driver);
    }

    public void logout() {
        click(menuButton);
        click(signOutButton);
    }

    public boolean isLoaded() {
        return isDisplayed(financialHealthTitle);
    }
}

File: e2e/pages/DashboardPage.js
export class DashboardPage {
  /**
   * @param {import('@playwright/test').Page} page
   */
  constructor(page) {
    this.page = page;
    this.menuButton = page.locator('button[aria-label="Open menu"]');
    this.signOutButton = page.locator('button:has-text("Sign Out")');
    this.financialHealthTitle = page.locator('text=Financial Health').first();
  }

  async logout() {
    await this.menuButton.click();
    await this.signOutButton.waitFor({ state: 'visible' });
    await this.signOutButton.click();
  }

  async isLoaded() {
    await this.financialHealthTitle.waitFor({ state: 'visible', timeout: 10000 });
    return await this.financialHealthTitle.isVisible();
  }
}
